exist-core/src/main/java/org/exist/xquery/pragmas/Optimize.java

```diff
@@ -261,22 +275,26 @@ public void after(XQueryContext context) throws XPathException {
     public void after(XQueryContext context, Expression expression) throws XPathException {
     }
 
-    private void addOptimizable(Optimizable optimizable) {final int axis = optimizable.getOptimizeAxis();
+    private void addOptimizable(Optimizable optimizable) {
+        final int axis = optimizable.getOptimizeAxis();
+
         if (!(axis == Constants.CHILD_AXIS || axis == Constants.SELF_AXIS || axis == Constants.DESCENDANT_AXIS ||
                 axis == Constants.DESCENDANT_SELF_AXIS || axis == Constants.ATTRIBUTE_AXIS ||
                 axis == Constants.DESCENDANT_ATTRIBUTE_AXIS)) {
             // reverse axes cannot be optimized
             return;
         }
+
         if (optimizables == null) {
             optimizables = new Optimizable[1];
             optimizables[0] = optimizable;
-        } else {
-            Optimizable o[] = new Optimizable[optimizables.length + 1];
-            System.arraycopy(optimizables, 0, o, 0, optimizables.length);
-            o[optimizables.length] = optimizable;
-            optimizables = o;
+            return;
         }
+
+        Optimizable[] o = new Optimizable[optimizables.length + 1];
+        arraycopy(optimizables, 0, o, 0, optimizables.length);
```

Comment:
By convention, this should be `System.arrayCopy` please.

Reply:
However, this could be further improved by refactoring to:

```java
optimizables  = Arrays.copyOf(optimizables, optimizables.length + 1);
optimizables[optimizables.length - 1] = optimizable;
```